go-controller/pkg/ovn/gateway.go

```diff
@@ -352,14 +463,20 @@ func (oc *DefaultNetworkController) gatewayInit(nodeName string, clusterIPSubnet
 			IPPrefix: gwLRPIP.String(),
 			Nexthop:  gwLRPIP.String(),
 		}
+		if gw.netInfo.IsSecondary() {
+			lrsr.ExternalIDs = map[string]string{
+				types.NetworkExternalID:  gw.netInfo.GetNetworkName(),
+				types.TopologyExternalID: gw.netInfo.TopologyType(),
+			}
+		}
 		p := func(item *nbdb.LogicalRouterStaticRoute) bool {
 			return item.IPPrefix == lrsr.IPPrefix &&
 				libovsdbops.PolicyEqualPredicate(lrsr.Policy, item.Policy)
 		}
-		err := libovsdbops.CreateOrReplaceLogicalRouterStaticRouteWithPredicate(oc.nbClient,
-			oc.GetNetworkScopedClusterRouterName(), &lrsr, p, &lrsr.Nexthop)
+		err := libovsdbops.CreateOrReplaceLogicalRouterStaticRouteWithPredicate(gw.nbClient,
+			gw.clusterRouterName, &lrsr, p, &lrsr.Nexthop)
```

Comment:
@qinqon it's failing here when adding the static route on L2. This is the error I get:
```
E0724 21:24:37.608376    1052 obj_retry.go:671] Failed to update *v1.Node, old=ovn-worker, new=ovn-worker, error: error creating gateway for node ovn-worker: failed to init shared interface gateway: error creating static route {UUID: BFD:<nil> ExternalIDs:map[k8s.ovn.org/network:tenantblue k8s.ovn.org/topology:layer2] IPPrefix:100.65.0.1 Nexthop:100.65.0.1 Options:map[] OutputPort:<nil> Policy:<nil> RouteTable:} in tenantblue_ovn_cluster_router: object not found
```

I forgot to "protect" the GW code creation - if we don't have a cluster router name (layer2) we should skip the operations involving it.